[eas-cli] Add update:revert-update-rollout command

wschurman opened the pull request:
<!-- If this PR requires a changelog entry, add it by commenting the PR with the command `/changelog-entry [breaking-change|new-feature|bug-fix|chore] [message]`. -->
<!-- You can skip the changelog check by labeling the PR with "no changelog". -->

# Why

Please describe the motivation for this PR, and link to relevant GitHub issues, forums posts, or feature requests.

# How

How did you build this feature or fix this bug and why?

# Test Plan

Please describe how you tested this change and how a reviewer could reproduce your test, especially if this PR does not include automated tests! If possible, please also provide terminal output and/or screenshots demonstrating your test/reproduction.

wschurman commented:
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/expo/eas-cli/3068?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#3068** <a href="https://app.graphite.dev/github/pr/expo/eas-cli/3068?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.dev/github/pr/expo/eas-cli/3068?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#3067** <a href="https://app.graphite.dev/github/pr/expo/eas-cli/3067?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* `main`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

quinlanj commented:
bonus points if we can have an fyi that describes an example. i think most people will struggle with understanding this unless if they are a power user 

<img width="1263" alt="Screenshot 2025-06-24 at 11 39 43 AM" src="https://github.com/user-attachments/assets/0dc6279b-ebfc-47ca-8c0d-116d6da8b406" />

wschurman commented:
> If this is too much work, an alternative solution is to choose branch, then choose runtime and then revert the rollout. We are guaranteed the rolled out update is the most recent on the branch, so it will be an easy lookup once the user can provide the runtime.

I like this idea! Will go forward with it.

wschurman commented:
Updated this to handle all rollout cases that I could think of, select by runtime version, and added an [FYI link](https://github.com/expo/fyi/pull/209). Thanks for the thorough review!

quinlanj reviewed (approved):
lgtm!

wschurman commented:
### Merge activity

* **Jul 7, 8:57 PM UTC**: A user started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/expo/eas-cli/3068).
